spec/features/add_peep_spec.rb

```diff
@@ -0,0 +1,20 @@
+RSpec.feature 'Add Peep' do 
+  context '#create' do 
+    let!(:user) { User.create({ firstname: "Carol", surname: "Gardiner", username: "cazzy", email: "caz@caz", password: "secret123" }) }
+    scenario 'peep button exists on user profile' do 
+      signin_and_submit()
+      expect(page.current_path).to eq "/profile/1"
+      expect(page).to have_button "PEEP"
+    end
```

Comment:
I love the testing here. You've been really thorough. A good refactor could be to remove some repeating tests. eg. You could add the`expect(page).to have_button "PEEP"` line to the test below and then completely remove this first test.